### PR TITLE
Restore and copy sourceimage

### PR DESCRIPTION
**Pull request by chregu**
- Added image:restore
- Added image:copy
- Added image:copy-all
- Fixed image:delete-all
- Deprecated image:clone
- Deprecated image:clone-all

### CODE REVIEW

File: src/Command/ImageCopyCommand.php [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+class ImageCopyCommand extends BaseRokkaCliCommand
+{
+    protected function configure()
+    {
+        $this
+            ->setName('image:copy')
+            ->setDescription('Copy the given image to another organization')
+            ->addArgument('hash', InputArgument::REQUIRED, 'The Source Image hash')
+            ->addArgument('destination', InputArgument::REQUIRED, 'The destination organization')
+            ->addOption('organization', null, InputOption::VALUE_REQUIRED, 'The organization to delete the images from')
```
Comment (dbu) [inline src/Command/ImageCopyCommand.php]: erm, why delete? should we call this option `source` instead? and mention that its for when you want to copy from a non-default organization?

File: src/Command/ImageRestoreCommand.php [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class ImageRestoreCommand extends BaseRokkaCliCommand
+{
+    protected function configure()
+    {
+        $this
+            ->setName('image:restore')
+            ->setDescription('Restore the given image from Rokka')
```
Comment (dbu) [inline src/Command/ImageRestoreCommand.php]: is "from" correct? maybe just "Restore an image"?

File: src/Command/ImageRestoreCommand.php [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+            ->addArgument('hash', InputArgument::REQUIRED, 'The Source Image hash')
+            ->addOption('organization', null, InputOption::VALUE_REQUIRED, 'The organization to delete the images from')
```
Comment (dbu) [inline src/Command/ImageRestoreCommand.php]: organization to delete the images from => organization in which to restore the image

(its singular here, right?)

File: src/Command/ImageRestoreCommand.php [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        if (!$client->restoreSourceImage($hash, $organization)) {
+            $output->writeln($this->formatterHelper->formatBlock([
+                'Error!',
+                'Error restoring the given image!',
+            ], 'error', true));
+
+            return -1;
+        }
+
+        $output->writeln('Image <info>'.$hash.'</info> restored from <comment>'.$organization.'</comment>.');
```
Comment (dbu) [inline src/Command/ImageRestoreCommand.php]: s/from/on/ ?

File: src/Command/ImageCopyCommand.php (hunk 1 above)
Comment (chregu) [reply src/Command/ImageCopyCommand.php]: it's organization in all other options with the almost the same meaning. I'd leave it like that.
The delete was a copy&paste mistake

Comment (dbu) [reply src/Command/ImageCopyCommand.php]: fair enough, lets keep it consistent then.

File: src/Command/ImageRestoreCommand.php (hunk 2 above)
Comment (chregu) [reply src/Command/ImageRestoreCommand.php]: removed. everywhere

File: src/Command/ImageRestoreCommand.php (hunk 3 above)
Comment (chregu) [reply src/Command/ImageRestoreCommand.php]: yep, fixed

File: src/Command/ImageRestoreCommand.php (hunk 4 above)
Comment (chregu) [reply src/Command/ImageRestoreCommand.php]: changed

**Comment by chregu** [comment]
I just realized there's an `image:clone` command already, which does the same as copy (just not serverside). Will rewrite..

File: src/Command/ImageCopyCommand.php (hunk 1 above)
Comment (chregu) [reply src/Command/ImageCopyCommand.php]: Just realized the image:clone command used it differently. 
```
            ->addArgument('dest-organization', InputArgument::REQUIRED, 'The destination organization to copy images to')
            ->addOption('source-organization', null, InputOption::VALUE_REQUIRED, 'The source organization to copy images from', null)
```

I'll therefore will use that for image:copy as well

**Comment by chregu** [comment]
fixed the stuff from the PR, made image:clone and image:clone-all deprecated and added image:copy-all

**Comment by dbu** [comment]
i think the clone-all was a first shot at being able to run a staging env and copy all images from production, and similar scenarios.

**Pull request merged**